### PR TITLE
Js integration testing

### PR DESCRIPTION
**Pull request by ajbraus**

### CODE REVIEW

File: w3_d4_2_js_integration_testing/README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+| :---: | :---: | :---: |
+| Integration Testing TDD/BDD | Mocha, Supertest, Chai | Challenges |
+
+### Preparatory Readings
+
+1. [Difference Between Test Methodologies](http://stackoverflow.com/questions/4904096/whats-the-difference-between-unit-functional-acceptance-and-integration-test/4904533#4904533)
+
+
+### Motivation (Why?)
+
```
Comment (janephilipps) [inline w3_d4_2_js_integration_testing/README.md:17]: are you planning to add something here?

File: w3_d4_2_js_integration_testing/README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+### Analogy (What?)
+
```
Comment (janephilipps) [inline w3_d4_2_js_integration_testing/README.md:21]: are you planning to add something here?

File: w3_d4_2_js_integration_testing/README.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+### Key Concepts & Snippets
+
+#### Unit & Integration Tests
+
+**Unit Tests** test only one part of the code independently.
+
+**Integration Tests** test all the code the way a user would use it.
+
+#### Example of a Simple API Test
+
```
Comment (janephilipps) [inline w3_d4_2_js_integration_testing/README.md:47]: Can you please add comments to this to explain what each part of the code is doing?

File: w3_d4_2_js_integration_testing/README.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+describe('home page should say "hello world!"', function() {
+  request(baseUrl, function (error, response, body) {
+    expect(body.to.contain('hello world!'));
+  });
+});
+```
+
+#### Example of Chai Expect Selectors
+
```
Comment (janephilipps) [inline w3_d4_2_js_integration_testing/README.md:97]: Please add an example with comments

File: w3_d4_2_js_integration_testing/README.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+#Challenges
+
+### Docs & Resources
+
+* **[Mocha](http://mochajs.org/#getting-started)** - for running asynchronous tests
+* **[Request](https://github.com/request)** - for handling HTTP request/response
+* **[Chai](http://chaijs.com/api/)** - for assertions
+
+### Basic Challenges
+
```
Comment (janephilipps) [inline w3_d4_2_js_integration_testing/README.md:136]: Can you be more specific here? Have you previously explained what a GET test is? I could see this being a little confusing for students if it's not super clear.

File: w3_d4_2_js_integration_testing/README.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+1. Add 3 GET tests to a project from this week
+2. Add 1 POST test to a project from this week
+
+### Stretch Challenges (View Testing)
+
+1. Add **[Cheerio](https://github.com/cheeriojs/cheerio)** to a project. Cheerio is for selecting HTML elements from responses (like "jQuery for tests")
+2. Test that the title of ```www.google.com``` is "Google".
+Hint:
```
Comment (janephilipps) [inline w3_d4_2_js_integration_testing/README.md]: Do we need to introduce another tool? This lesson already has 3 new tools.

**Comment by apennell** [comment]
Looks like a good start, but it isn't finished, is it?

**Comment by ajbraus** [comment]
correct

Adam Braus
608-770-0230

**Comment by ajbraus** [comment]
I think I got things good now. 

Example with psuedo-code CHECK!
A working example CHECK!
FREAKING AWESOME CHECK!

**Comment by ben-manning** [comment]
Hey, really quickly, on line 27 what does that medium testing coverage cover 30% of?

Everything else looks pretty clear.

**Pull request merged**